go/obscuronode/host/main/main.go

```diff
@@ -22,7 +23,13 @@ func main() {
 	hostCfg := host.AggregatorCfg{GossipRoundDuration: time.Duration(*config.gossipRoundNanos), ClientRPCTimeoutSecs: *config.rpcTimeoutSecs}
 	enclaveClient := host.NewEnclaveRPCClient(*config.enclaveAddr, host.ClientRPCTimeoutSecs*time.Second, nodeID)
 	aggP2P := p2p.NewSocketP2PLayer(*config.ourP2PAddr, config.peerP2PAddrs)
-	agg := host.NewObscuroAggregator(nodeID, hostCfg, l1NodeDummy{}, nil, *config.isGenesis, enclaveClient, aggP2P)
+	w := wallet.NewInMemoryWallet(*config.privateKeyString)
+	contractAddr := common.HexToAddress(*config.contractAddress)
+	l1Client, err := l1client.NewEthClient(nodeID, "127.0.0.1", 7545, w, contractAddr)
```

Comment:
@otherview For me, this is another strong argument for not making the wallet an argument to the Ethereum client. It would be better if the host didn't need a wallet and private key just to query for txs, etc.

Reply:
Yeah, good point, I share the same thought. originally I decoupled the eth client and the wallet however that has a bit more deep impact because of the interface provided methods. Stuff like `BroadcastTx(t *obscurocommon.L1TxData) ` has to change quite a bit right ?

I was planning on leaving that to another interation, unless you guys want me to change it now ?

https://github.com/obscuronet/obscuro-playground/blob/eth_nodes/go/l1client/ethereum_client.go#L24-L25

I'm not following the host part, can you expand on that please ?

Reply:
@otherview Unless I'm mistaken, the host isn't producing any transactions, right? They're created by the enclave. So why is the host being handed a structure that contains the private keys? It feels like v bad infosec.

Reply:
The enclave is not producing ethereum transactions, it's producing rollups.
So the enclave is agnostic to eth transactions. It just needs to produce good rollups (with the attested code).
The host needs the private keys to sign the transactions. I don't see the infosec issue.